File: solutions/hamming/Python/hamming.py
def distance(strand_a, strand_b):
    if len(strand_a) != len(strand_b):
        raise ValueError("Strands must be of equal length.")

    return sum(i != j for i, j in zip(strand_a, strand_b))

#    hamming_dist = 0
#    for i in range(len(strand_a)):
#        if strand_a[i] != strand_b[i]:
#            hamming_dist += 1
#    return hamming_dist

File: solutions/hamming/PHP/Hamming.php
<?php

declare(strict_types=1);

function distance(string $strandA, string $strandB): int
{
    if (strlen($strandA) !== strlen($strandB)) {
        throw new InvalidArgumentException("DNA strands must be of equal length.");
    }

    $dist = 0;
    foreach (mb_str_split($strandA) as $i => $cA) {
        if ($strandB[$i] != $cA) {
            $dist++;
        }
    }
    return $dist;
}
